### PR TITLE
Technical Documentation Page Self Review

### PR DESCRIPTION
**Pull request by Shewolf227**
Learning Objectives:

- Apply best practices in HTML code.
- Use semantic HTML tags.
- Use CSS selectors correctly.
- Use CSS box model.
- Perform a code review for yourself on a past project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# FCC Test Suite Template
```
Comment (Shewolf227) [inline README.md:1]: I think adding a custom ReadMe that is related to this project is better than having this generic type of ReadMe

File: css/style.css [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#navbar {
+	width: 100%;
+	border-bottom: solid;
+	border-color: rgba(0, 22, 22, 0.4);
+	display: flex;
+	flex-wrap: wrap;
+	justify-content: center;
+	align-items: start;
+	padding-bottom: 20px;
+}
```
Comment (Shewolf227) [inline css/style.css:10]: I think adding an empty line between css class declarations makes the code easier to easier.

File: css/style.css [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+#main-header {
+	display: block;
+	width: 100%;
+	text-align: center;
+}
+#main-doc {
```
Comment (Shewolf227) [inline css/style.css:16]: I think adding an empty line between css class declarations makes the code easier to easier.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,143 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<meta charset="UTF-8" />
+		<title>Fork Me! FCC: Test Suite Template</title>
```
Comment (Shewolf227) [inline index.html:5]: I think, changing the `title` of the webpage adds more personality to the webpage

File: index.html [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+				<header id="main-header">REACT DOCUMENTATION</header>
+				<a class="nav-link" href="#Hello_World">Hello World</a>
+				<a class="nav-link" href="#Introducing_JSX">Introducing JSX</a>
+				<a class="nav-link" href="#Rendering_Elements">Rendering Elements</a>
+				<a class="nav-link" href="#Components_and_Props">Components and Props</a>
+				<a class="nav-link" href="#State_and_Lifecycle">State and Lifecycle</a>
+			</nav>
+			<div class="container">
+				<section id="Hello_World" class="main-section">
+					<header>Hello World</header>
```
Comment (Shewolf227) [inline index.html:21]: I think this should be changed to `<h1>` instead, for semantic purposes

File: index.html [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+					<p>The smallest React example looks like this:</p>
+					<code>
+						ReactDOM.render(
+						<span>Hello, world!</span>
+						, document.getElementById('root') );
+					</code>
+					<p>It displays a heading saying “Hello, world!” on the page.</p>
+				</section>
+				<section id="Introducing_JSX" class="main-section">
+					<header>Introducing JSX</header>
```
Comment (Shewolf227) [inline index.html:31]: I think this should be changed to `<h1>` instead, for semantic purposes

File: index.html [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+					<p>
+						This funny tag syntax is neither a string nor HTML. It is
+						called JSX, and it is a syntax extension to JavaScript. We
+						recommend using it with React to describe what the UI should
+						look like. JSX may remind you of a template language, but it
+						comes with the full power of JavaScript.
+					</p>
+				</section>
+				<section id="Rendering_Elements" class="main-section">
+					<header>Rendering Elements</header>
```
Comment (Shewolf227) [inline index.html:43]: I think this should be changed to `<h1>` instead, for semantic purposes

File: index.html [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+					</p>
+					<code>const element = <span>Hello, world</span>;</code>
+					<p>
+						Unlike browser DOM elements, React elements are plain
+						objects, and are cheap to create. React DOM takes care of
+						updating the DOM to match the React elements.
+					</p>
+				</section>
+				<section id="Components_and_Props" class="main-section">
+					<header>Components and Props</header>
```
Comment (Shewolf227) [inline index.html:56]: I think this should be changed to `<h1>` instead, for semantic purposes

File: index.html [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+						>function Welcome(props) { return
+						<span>Hello, {props.name}</span>; }</code
+					>
+					<p>
+						The simplest way to define a component is to write a
+						JavaScript function:
+					</p>
+				</section>
+				<section  id="State_and_Lifecycle" class="main-section">
+					<header>State and Lifecycle</header>
```
Comment (Shewolf227) [inline index.html:76]: I think this should be changed to `<h1>` instead, for semantic purposes

File: js/index.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  - As you start to build out your 
+    project, when tests are failing, 
+    you should get helpful errors 
+    along the way!
+    ************/
+
+// PLEASE NOTE: Adding global style rules using the * selector, or by adding rules to body {..} or html {..}, or to all elements within body or html, i.e. h1 {..}, has the potential to pollute the test suite's CSS. Try adding: * { color: red }, for a quick example!
+
+// Once you have read the above messages, you can delete all comments. 
+"use strict";
```
Comment (Shewolf227) [inline js/index.js:28]: Leaving an empty line at the end of a file is one of the best practices that you might want to adopt

File: README.md (hunk 1 above)
Comment (xtrahuman) [reply README.md:1]: Nice review, but it is missing the final summary. 

Please check this link on [how to summarize your review](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/code-review/articles/give_code_review_basics.md)

Also, Kindly make sure to drop at least two inline comment on this project

**Review by xtrahuman — CHANGES_REQUESTED** [review]
# Required Changes ♻️

Hello @SoniaKarungi, 

You have done a great job so far! 😃 

But there is a little issue stopping this project from getting approved.

image

_Kindly check the comments under the review._ 👇  👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (Shewolf227) [reply README.md:1]: Hi @xtrahuman Thanks for your feedback. However, I don't quite understand what you mean by "it is missing the final summary.". Could you please help clarify this?

Thank you

**Review by Shewolf227 — COMMENTED** [review]
Hi @SoniaKarungi, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

File: css/style.css (hunk 2 above)
Comment (topeogunleye) [reply css/style.css:10]: Nice job being polite, giving a reason for the requested change, and phrasing your comments as suggestions

File: css/style.css (hunk 3 above)
Comment (topeogunleye) [reply css/style.css:16]: Nice job being polite, giving a reason for the requested change, and phrasing your comments as suggestions

File: index.html (hunk 4 above)
Comment (topeogunleye) [reply index.html:5]: Nice job being polite, giving a reason for the requested change, and phrasing your comments as suggestions

File: index.html (hunk 5 above)
Comment (topeogunleye) [reply index.html:21]: Nice job being polite, giving a reason for the requested change, and phrasing your comments as suggestions

File: index.html (hunk 6 above)
Comment (topeogunleye) [reply index.html:31]: Nice job being polite, giving a reason for the requested change, and phrasing your comments as suggestions

**Review by topeogunleye — APPROVED** [review]
### Hi @SoniaKarungi,
Great Job on this project review! 👍🏻
# Status: Approved ✔️
Excellent job on implementing all the changes requested. Keep it up 👏
Your project review is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

### To Highlight
- [x]  Added descriptive inline comments.
- [x]  Included an explanation, a "why", for each requested change.
- [x]  Great use of emojis to make your code review summary more human-friendly 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Congratulations-Memes-FEATURED

**Pull request merged**